app/src/main/java/com/example/weather/network/Dao.kt

```diff
@@ -0,0 +1,31 @@
+package com.example.weather.network
+
+import androidx.room.Dao
+import androidx.room.Insert
+import androidx.room.Query
+
+@Dao
+interface CityOverviewDao {
+
+    @Query("SELECT * FROM city_overview_table where name = name")
+    fun getCities(): List<CachedCityOverview>
+
+    @Insert
```

Comment:
we need to handle conflicts when data is inserted into the table. try this 
`@Insert(onConflict = OnConflictStrategy.REPLACE)
`